modules/apps/depot/depot-service/src/main/java/com/liferay/depot/service/impl/DepotEntryServiceImpl.java

```diff
@@ -67,6 +67,24 @@ public DepotEntry deleteDepotEntry(long depotEntryId)
 		return depotEntryLocalService.deleteDepotEntry(depotEntryId);
 	}
 
+	@Override
+	public List<DepotEntry> getCurrentAndGroupConnectedDepotEntries(
+			long groupId, int start, int end)
+		throws PortalException {
+
+		List<DepotEntry> filteredDepotEntries = getGroupConnectedDepotEntries(
+			groupId, start, end);
+
+		DepotEntry depotEntry = depotEntryLocalService.fetchGroupDepotEntry(
+			groupId);
+
+		if (depotEntry != null) {
+			filteredDepotEntries.add(depotEntry);
+		}
+
+		return filteredDepotEntries;
+	}
```

Comment:
Cannot we change the original method (`getGroupConnectedDepotEntries`)? It seems reasonable to always see a Depot as connected to itself.

Reply:
I'm not sure about it
The method is called from several places and some of them I do not trigger them so I preferred to create the new method following the pattern as the _getCurrentAndAncestorSiteGroups_
Do you think that is worth the risk?

(if **yes**, I will change the LocalService instead of the service.)

Reply:
That's fine. I'm ok with the changes then.